batcher/aligned-batcher/src/lib.rs

```diff
@@ -1189,39 +1192,62 @@ impl Batcher {
         proof_submitters: Vec<Address>,
         fee_params: CreateNewTaskFeeParams,
     ) -> Result<TransactionReceipt, BatcherError> {
-        info!("Creating task for: 0x{}", hex::encode(batch_merkle_root));
-
-        match try_create_new_task(
-            batch_merkle_root,
-            batch_data_pointer.clone(),
-            proof_submitters.clone(),
-            fee_params.clone(),
-            &self.payment_service,
-        )
-        .await
-        {
-            Ok(receipt) => Ok(receipt),
-            Err(BatcherSendError::TransactionReverted(err)) => {
-                // Since transaction was reverted, we don't want to retry with fallback.
-                warn!("Transaction reverted {:?}", err);
-
-                Err(BatcherError::TransactionSendError)
-            }
-            Err(_) => {
-                let receipt = try_create_new_task(
+        let result = retry_function(
+            || {
+                create_new_task_retryable(
                     batch_merkle_root,
-                    batch_data_pointer,
-                    proof_submitters,
-                    fee_params,
+                    batch_data_pointer.clone(),
+                    proof_submitters.clone(),
+                    fee_params.clone(),
+                    &self.payment_service,
                     &self.payment_service_fallback,
                 )
-                .await?;
-
-                Ok(receipt)
+            },
+            DEFAULT_MIN_RETRY_DELAY,
+            DEFAULT_BACKOFF_FACTOR,
+            DEFAULT_MAX_RETRIES,
+        )
+        .await;
+        match result {
+            Ok(receipt) => Ok(receipt),
+            Err(RetryError::Permanent(BatcherError::ReceiptNotFoundError)) => {
+                self.metrics.canceled_batches.inc();
+                self.cancel_create_new_task_tx(fee_params.gas_price).await;
+                Err(BatcherError::ReceiptNotFoundError)
             }
+            Err(_) => Err(BatcherError::TransactionSendError),
         }
     }
 
+    // Sends a transaction to Ethereum with the same nonce as the previous one to override it.
+    // We use the maximum number of retries here because this transaction must be included.
+    pub async fn cancel_create_new_task_tx(&self, old_task_gas_price: U256) {
+        info!("Canceling created task");
```

Comment:
logging style nit :)
```suggestion
        info!("Cancelling createNewTask transaction...");
```